cdap-app-templates/cdap-etl/cdap-calcite/src/main/java/io/cdap/cdap/datapipeline/relational/CDAPCalciteSchema.java

```
@@ -0,0 +1,235 @@
+        String fieldName = cdapField.getName().toUpperCase();
+
+        RelDataType calciteDatatype = getDatatypeFromSchema(cdapSchema);
+        RelDataTypeFieldImpl calciteField = new RelDataTypeFieldImpl(fieldName, index, calciteDatatype);
+
+        return calciteField;
+    }
+
+    private static RelDataType getDatatypeFromSchema(Schema cdapSchema) {
+        try {
+            return getDatatypeFromSchema(cdapSchema, false);
+        } catch (SQLDialectException e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    private static RelDataType getDatatypeFromSchema(Schema cdapSchema, boolean isNullable)
+            throws SQLDialectException {
+        Schema.Type cdapDatatype = cdapSchema.getType();
+        RelDataType calciteDatatype;
+
+        switch (cdapDatatype) {
+            case NULL:
+                calciteDatatype = new BasicSqlType(RelDataTypeSystem.DEFAULT, SqlTypeName.NULL);
+                break;
+            case BOOLEAN:
+                calciteDatatype = new BasicSqlType(RelDataTypeSystem.DEFAULT, SqlTypeName.BOOLEAN);
+                break;
+            case INT:
+                calciteDatatype = getDatatypeFromInt(cdapSchema);
+                break;
+            case LONG:
+                calciteDatatype = getDatatypeFromLong(cdapSchema);
+                break;
+            case FLOAT:
+                calciteDatatype = new BasicSqlType(RelDataTypeSystem.DEFAULT, SqlTypeName.FLOAT);
+                break;
+            case DOUBLE:
+                calciteDatatype = new BasicSqlType(RelDataTypeSystem.DEFAULT, SqlTypeName.DOUBLE);
+                break;
+            case BYTES:
+                calciteDatatype = getDatatypeFromBytes(cdapSchema);
+                break;
+            case STRING:
+                calciteDatatype = getDatatypeFromString(cdapSchema);
+                break;
+            case ENUM:
+                // TODO: Add implementation
+                calciteDatatype = null;
```

Please throw UnsupportedOperationException for anything that is not supported